src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+  }
+
+  applyDamage(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health < 0) {
```

Comment:
[CODE PERFORMANCE] The `applyDamage` method uses `Animal.removeDead()` inside a conditional that checks if the health is below 0. However, this could lead to unnecessary executions of `Animal.removeDead` if multiple damage applications occur before the animal array is filtered. It would be better to call `Animal.removeDead` after all damage has been applied and not within the `applyDamage` method.